.github/workflows/non-target-specific-extended.yml

```diff
@@ -0,0 +1,61 @@
+name: Non-target-specific extended
+
+on:
+  workflow_dispatch:
+  workflow_call:
+    inputs:
+      all:
+        type: boolean
+  pull_request:
+    paths-ignore:
+    - 'org.lflang/src/org/lflang/generator/c/**'
+    - 'org.lflang/src/org/lflang/generator/cpp/**'
+    - 'org.lflang/src/org/lflang/generator/python/**'
+    - 'org.lflang/src/org/lflang/generator/rust/**'
+    - 'org.lflang/src/org/lflang/generator/ts/**'
+    - 'org.lflang/src/lib/c/**'
+    - 'org.lflang/src/lib/platform/**'
+    - 'org.lflang/src/lib/cpp/**'
+    - 'org.lflang/src/lib/py/**'
+    - 'org.lflang/src/lib/rs/**'
+    - 'org.lflang/src/lib/ts/**'
+
+env:
+  # 2020.11
+  vcpkgGitRef: 0bf3923f9fab4001c00f0f429682a0853b5749e0
+
+concurrency:
+  group: only-non-target-specific-extended-${{ github.ref }}-${{ github.event_path }}
+  cancel-in-progress: ${{ github.ref != 'refs/heads/master' }}
```

Comment:
I think this means that if the job is run from `master`, it won't be canceled, but this suggests that jobs from subsequent while the PR is not in draft mode, will not get cancelled. This is different from the current behavior. I think they _should_ be canceled in that situation...

This relates to my earlier point about dealing with canceled workflows when switching from "draft" to "ready-for-review" while tests are running. I think that we should pass into the `env` whether the workflow is `catch-all` or `only` and use that in the definition of the `group`. I think that we could then just have `cancel-in-progress` set to `true` and achieve the desired behavior; `only` jobs will not be canceled if a `catch-all` job is spawned (meaning it won't invalidate the end result, as it does now), but subsequent `catch-all` jobs will cancel each other.

Reply:
OK. I think that makes sense.

An alternative solution (which might solve other problems as well) would be to just have the `catch-all` workflows be unconditionally triggered, and let them conditionally trigger the more targeted jobs. It would do this by using `git` to determine which files are different from in master rather than by using the special workflow trigger.

Reply:
> An alternative solution (which might solve other problems as well) would be to just have the `catch-all` workflows be unconditionally triggered, and let them conditionally trigger the more targeted jobs. It would do this by using `git` to determine which files are different from in master rather than by using the special workflow trigger.

I'm fine with either approach, but, indeed, I think the approach you outline here would have advantages; we won't have any "skipped" workflows anymore.

Reply:
We could even do fancier things when we don't rely on the built-in GH Actions feature; we could e.g., skip doing CI altogether when only a `.md` file is edited.

Reply:
Great! I think that this more elaborate approach will work.

Reply:
It works beautifully!